vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -179,21 +187,52 @@ public ISpace getFullyLoadedSpace(String id) {
         return spaces;
     }
 
-    /**
-     * Method to delete space based on id
-     * 
-     * @param id
-     *            if id is null throws exception, else delete corresponding
-     *            space
-     * @throws SpaceDoesNotExistException 
-     */
-    @Override
-    public void deleteSpaceById(String id) throws SpaceDoesNotExistException {
-        try {
-            spaceRepo.deleteById(id);
-        } catch (IllegalArgumentException | EmptyResultDataAccessException exception) {
-            throw new SpaceDoesNotExistException(exception);
-        }
-
-    }
+	/**
+	 * Method to delete space based on id
+	 * 
+	 * @param id if id is null throws exception, else delete corresponding space
+	 * @throws SpaceDoesNotExistException
+	 */
+	@Override
+	public void deleteSpaceById(String id) throws SpaceDoesNotExistException {
+		try {
+			// Order of deletion: Tables: SPACELINKDISPLAY (link_id), SPACELINK
+			// (source_space_id), SPACEDISPLAY (space_id), SPACE(id)
+			List<String> linkedIds = null;
+			List<SpaceLink> spaceLinkIDs = spaceLinkRepo.getLinkedSpaceIds(id);
+
+			// When the space has other links attached to it
+			if (spaceLinkIDs.size() > 0) {
+				linkedIds = new ArrayList<>();
+				for (SpaceLink spaceData : spaceLinkIDs) {
+					linkedIds.add(spaceData.getId());
+				}
+				spaceLinkDisplayRepo.deleteByLinkId(linkedIds);
+				spaceLinkRepo.deleteBySourceSpaceId(id);
+				spaceDisplayRepo.deleteBySpaceId(id);
+				spaceRepo.deleteById(id);
+			}
+			// When the space has no other links attached to it
+			else {
+				spaceDisplayRepo.deleteBySpaceId(id);
+				spaceRepo.deleteById(id);
+			}
+
+		} catch (IllegalArgumentException | EmptyResultDataAccessException exception) {
+			throw new SpaceDoesNotExistException(exception);
+		}
+
+	}
+
+	@Override
+	public boolean checkTargetSpaceIds(String id) {
+
+		List<SpaceLink> getLinksFromSpace = spaceLinkRepo.getLinkedSpaceIds(id);
```

Comment:
bad variable name, you don't start variables with `get` usually (that's for methods)